tests/test.py

```diff
@@ -0,0 +1,204 @@
+        try:
+            output = subprocess.check_output(args)
+            exitcode = 0
+        except subprocess.CalledProcessError, e:
+            output = e.output
+            exitcode = e.returncode
+        print '>> %s (exit code %s, expected %s)\n%s' % (args, exitcode, expected_exit_code, output)
+        assert expected_exit_code == exitcode, 'Exit codes don\'t match'
+        return output.rstrip()
+
+    def setUp(self):
+        self.browser = webdriver.Firefox()
+        self.addCleanup(self.browser.quit)
+        self.base_url = "http://localhost:8000/"
+
+    def testBaseURL(self):
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+
+        # check title
+        self.assertIn('CodaLab - Home', self.browser.title)
+
+        #check the list in the top right
+        for i, item in [(1, 'Worksheets'), (2, 'Competitions'), (3, 'Help')]:
+            self.assertEqual(
+                item,
+                self.browser.find_element_by_xpath("html/body/nav/div/div[2]/ul/li[" + str(i) + "]/a").get_attribute('innerHTML')
+            )
+
+        for i, item in [(1, 'Worksheets'), (2, 'Competitions')]:
+            self.assertEqual(
+                item,
+                self.browser.find_element_by_xpath("html/body/div[2]/div/div/div[" + str(i) + "]/div/div/a").get_attribute('innerHTML')
+            )
+
+        assert "Run reproducible experiments and created executable papers using worksheets" in self.browser.page_source
+        assert "Enter an existing competition to solve challenging data problems, or host your own." in self.browser.page_source
+
+        for i, item in ([(3, 'Example Worksheets'), (4, 'Latest Competitions')]):
+            self.assertEqual(
+                item,
+                self.browser.find_element_by_xpath("html/body/div[2]/div/div/div[" + str(i) + "]/h2").get_attribute('innerHTML')
+            )
+
+    def testWorksheet(self):
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+        self.browser.find_element_by_xpath("html/body/nav/div/div[2]/ul/li[1]/a").click()
+        self.assertEqual('Worksheets',
+            self.browser.find_element_by_xpath("html/body/div[1]/div/h1").get_attribute('innerHTML')
+        )
+
+    def testWorksheetContent(self):
+        cl = 'cl'
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+        
+        # create a temp worksheet for testing
+        temp_worksheet_name = 'test-codalab-' + str(random.randint(0, 1000000))
+        temp_worksheet_title =  'The really long testing title for testing the automated testing of worksheets'
+        temp_worksheet_uuid = self.run_command([cl, 'new', temp_worksheet_name])
+        temp_file_name = 'cl-web-interface-testing.txt'
+        temp_worksheet_paragraph_content = "For running automated testing on the cli " * 10
+        self.run_command([cl, 'work', temp_worksheet_uuid])
```

Comment:
It'd be better if you were running these commands through the action bar on the website...is this easy to do?